### ISSUE TITLE
Improve the custom loss

### ISSUE THREAD

**Issue by eduardocorrearaujo** [eduardocorrearaujo/lstm_transf_to_state#1]
In this [notebook](https://github.com/eduardocorrearaujo/lstm_transf_to_state/blob/main/custom_loss_model.ipynb) there is the implementation of a custom loss function defined as: 
```
def custom_loss(y_true, y_pred):
    '''
    Test of a custom loss function
    '''
    first_log = tf.math.log(y_pred + 1.0)
    second_log = tf.math.log(y_true + 1.0)

    sum_values = abs(tf.reduce_sum(y_pred) - tf.reduce_sum(y_true))

    peak_mag = abs(tf.reduce_max(y_pred) - tf.reduce_max(y_true))

    peak_pos = abs(tf.argmax(y_pred) - tf.argmax(y_true))

    sum_values = tf.cast(sum_values, tf.float32)
    peak_mag = tf.cast(peak_mag, tf.float32)
    peak_pos = tf.cast(peak_pos, tf.float32)

    msle_error = tf.reduce_mean(tf.square(first_log - second_log))

    return msle_error + 0.1*sum_values + 0.1*peak_mag + 0.2*peak_pos
```

The idea behind this loss function is to create a loss capable of considering not just a mean performance as `mse` and `msle` metrics but also considering important epidemiological aspects, the total sum of cases (`sum_values`), the date of peak  (`peak_pos`) and the number of cases in the peak (`peak_mag`). 

The problem is that this implementation is not allowing the network to learn the pattern of the data since the output is random peaks:

![Captura de Tela 2024-07-22 às 22 41 05](https://github.com/user-attachments/assets/6994a601-bc6a-4805-adb4-ac3d3107752d)

I would like to have help with the implementation of this custom loss to understand if I did some step wrong or if the implementation of the network must be changed to use this custom loss.

**Comment by eduardocorrearaujo** [comment]
In this [notebook](https://github.com/eduardocorrearaujo/lstm_transf_to_state/blob/main/custom_loss_model.ipynb) there is the implementation of a custom loss function defined as: 
```
def custom_loss(y_true, y_pred):
    '''
    Test of a custom loss function
    '''
    first_log = tf.math.log(y_pred + 1.0)
    second_log = tf.math.log(y_true + 1.0)

    sum_values = abs(tf.reduce_sum(y_pred) - tf.reduce_sum(y_true))

    peak_mag = abs(tf.reduce_max(y_pred) - tf.reduce_max(y_true))

    peak_pos = abs(tf.argmax(y_pred) - tf.argmax(y_true))

    sum_values = tf.cast(sum_values, tf.float32)
    peak_mag = tf.cast(peak_mag, tf.float32)
    peak_pos = tf.cast(peak_pos, tf.float32)

    msle_error = tf.reduce_mean(tf.square(first_log - second_log))

    return msle_error + 0.1*sum_values + 0.1*peak_mag + 0.2*peak_pos
```

The idea behind this loss function is to create a loss capable of considering not just a mean performance as `mse` and `msle` metrics but also considering important epidemiological aspects, the total sum of cases (`sum_values`), the date of peak  (`peak_pos`) and the number of cases in the peak (`peak_mag`). 

The problem is that this implementation is not allowing the network to learn the pattern of the data since the output is random peaks:

![Captura de Tela 2024-07-22 às 22 41 05](https://github.com/user-attachments/assets/6994a601-bc6a-4805-adb4-ac3d3107752d)


I would like to have help with the implementation of this custom loss to understand if I did some step wrong or if the implementation of the network must be changed to use this custom loss.
